Oracle Vault panics whenever ExternalSecrets are added

Hi,

Sorry if I skipped over the issue format - it looks more like a feature request then a bug report one.

Anyways, I've been trying to setup the Oracle Vault provider for 2 hours now and keep causing the pod to panic.

Ive tried the following on both v0.3.8 and v0.3.6, getting the same `panic: reflect: call of reflect.Value.IsNil on struct Value` error no-matter what I try.

Steps to replicate:
1. Install helm chart in its own namespace (`helm install external-secrets/external-secrets -n external-secrets`)
2. Add a Secret containing both privateKey (downloaded from the oracle cloud website and is in pem format) and the fingerprint.
Below are examples of information put in
```
Fingerprint: 52:51:b7:79:8c:72:f4:54:16:d6:xx:xx:xx:xx:xx:xx
Private Key:
-----BEGIN PRIVATE KEY-----
/// redacted
-----END PRIVATE KEY-----
```
3. Add a SecretStore:
```yaml
kind: SecretStore
apiVersion: external-secrets.io/v1alpha1
metadata:
  name: oracle-cloud
spec:
  provider:
    oracle:
      auth:
        secretRef:
          fingerprint:
            key: fingerprint
            name: oracle-secret
          privatekey:
            key: privateKey
            name: oracle-secret
      region: ap-sydney-1
      tenancy: >-
        ocid1.tenancy.oc1..aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
      user: >-
        ocid1.user.oc1..aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```
4. Add an ExternalSecret:
```yaml
apiVersion: external-secrets.io/v1alpha1
kind: ExternalSecret
metadata:
  name: vault-secrets
spec:
  refreshInterval: 5m
  secretStoreRef:
    name: oracle-cloud
    kind: SecretStore
  target:
    name: vault-secrets
    creationPolicy: Owner
  data:
  - secretKey: my_secret
    remoteRef:
      key: ocid1.vaultsecret.oc1.ap-sydney-1.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```
5. Observe the pod in the `external-secrets` namespace crash and enter a CrashLoopBackOff status, erroring with:
(note this log is from v0.3.6, however it looks identical on v0.3.8)
```
panic: reflect: call of reflect.Value.IsNil on struct Value

goroutine 346 [running]:
reflect.Value.IsNil(...)
        /opt/hostedtoolcache/go/1.16.8/x64/src/reflect/value.go:1086
github.com/external-secrets/external-secrets/pkg/utils.IsNil(0x223e1e0, 0x4000438000, 0x50)
        /home/runner/work/external-secrets/external-secrets/pkg/utils/utils.go:42 +0x19c
github.com/external-secrets/external-secrets/pkg/provider/oracle.(*VaultManagementService).GetSecret(0x400044afd0, 0x27438f0, 0x40006601b0, 0x4000ecc180, 0x5e, 0x0, 0x0, 0x0, 0x0, 0x14, ...)
        /home/runner/work/external-secrets/external-secrets/pkg/provider/oracle/oracle.go:126 +0x84
github.com/external-secrets/external-secrets/pkg/controllers/externalsecret.(*Reconciler).getProviderSecretData(0x4000f34480, 0x27438f0, 0x40006601b0, 0x2731740, 0x400044afd0, 0x40002a4780, 0x0, 0x21, 0x22f14a4)
        /home/runner/work/external-secrets/external-secrets/pkg/controllers/externalsecret/externalsecret_controller.go:371 +0x194
github.com/external-secrets/external-secrets/pkg/controllers/externalsecret.(*Reconciler).Reconcile.func3(0x0, 0x0)
        /home/runner/work/external-secrets/external-secrets/pkg/controllers/externalsecret/externalsecret_controller.go:200 +0x9c
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0x4000470a00, 0x400011654a, 0x6, 0x40001175b0, 0xe, 0x2776068, 0x4000466b40, 0x2776068, 0x4000466b40)
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/controller/controllerutil/controllerutil.go:339 +0x28
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x27438f0, 0x40006601b0, 0x275f848, 0x4000ec4e10, 0x2776068, 0x4000466b40, 0x4000470a00, 0x0, 0x400011654a, 0x6, ...)
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/controller/controllerutil/controllerutil.go:201 +0x150
github.com/external-secrets/external-secrets/pkg/controllers/externalsecret.(*Reconciler).Reconcile(0x4000f34480, 0x27438f0, 0x40006601b0, 0x400011654a, 0x6, 0x4000ebf8d8, 0x14, 0x4000660100, 0x0, 0x0, ...)
        /home/runner/work/external-secrets/external-secrets/pkg/controllers/externalsecret/externalsecret_controller.go:221 +0x9f8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0x4000f23400, 0x2743848, 0x4000f34600, 0x1eccb60, 0x4000e2c020)
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:298 +0x268
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0x4000f23400, 0x2743848, 0x4000f34600, 0x40006bea00)
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:253 +0x1c4
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0x4000f0f0c0, 0x4000f23400, 0x2743848, 0x4000f34600)
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:214 +0x60
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:210 +0x34c
```

Help would be much appreciated.
Cheers.

cc: @KianTigger

--- a/pkg/utils/utils.go
+++ b/pkg/utils/utils.go
@@ -39,7 +39,14 @@ func MergeStringMap(dest, src map[string]string) {
 
 // IsNil checks if an Interface is nil.
 func IsNil(i interface{}) bool {
-	return i == nil || reflect.ValueOf(i).IsNil()
+	if i == nil {
+		return true
+	}
+	value := reflect.ValueOf(i)
+	if value.Type().Kind() == reflect.Ptr {
+		return value.IsNil()
+	}
+	return false
 }
 
 // ObjectHash calculates md5 sum of the data contained in the secret.

--- a/pkg/utils/utils_test.go
+++ b/pkg/utils/utils_test.go
@@ -17,6 +17,7 @@ package utils
 import (
 	"testing"
 
+	vault "github.com/oracle/oci-go-sdk/v45/vault"
 	v1 "k8s.io/api/core/v1"
 )
 
@@ -56,6 +57,91 @@ func TestObjectHash(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			if got := ObjectHash(tt.input); got != tt.want {
 				t.Errorf("ObjectHash() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestIsNil(t *testing.T) {
+	tbl := []struct {
+		name string
+		val  interface{}
+		exp  bool
+	}{
+		{
+			name: "simple nil val",
+			val:  nil,
+			exp:  true,
+		},
+		{
+			name: "nil slice",
+			val:  (*[]struct{})(nil),
+			exp:  true,
+		},
+		{
+			name: "struct pointer",
+			val:  &testing.T{},
+			exp:  false,
+		},
+		{
+			name: "struct",
+			val:  testing.T{},
+			exp:  false,
+		},
+		{
+			name: "slice of struct",
+			val:  []struct{}{{}},
+			exp:  false,
+		},
+		{
+			name: "slice of ptr",
+			val:  []*testing.T{nil},
+			exp:  false,
+		},
+		{
+			name: "slice",
+			val:  []struct{}(nil),
+			exp:  false,
+		},
+		{
+			name: "int default value",
+			val:  0,
+			exp:  false,
+		},
+		{
+			name: "empty str",
+			val:  "",
+			exp:  false,
+		},
+		{
+			name: "oracle vault",
+			val:  vault.VaultsClient{},
+			exp:  false,
+		},
+		{
+			name: "func",
+			val: func() {
+				// noop for testing and to make linter happy
+			},
+			exp: false,
+		},
+		{
+			name: "channel",
+			val:  make(chan struct{}),
+			exp:  false,
+		},
+		{
+			name: "map",
+			val:  map[string]string{},
+			exp:  false,
+		},
+	}
+
+	for _, row := range tbl {
+		t.Run(row.name, func(t *testing.T) {
+			res := IsNil(row.val)
+			if res != row.exp {
+				t.Errorf("IsNil(%#v)=%t, expected %t", row.val, res, row.exp)
 			}
 		})
 	}

